tme/service_test.go

```diff
@@ -0,0 +1,276 @@
+		svc := NewService(repos, tmpfile.Name(), successHTTPclient, "/base/url", 1, "/endpoint", 1)
+		count, err := svc.GetCount("topics")
+		assert.Error(t, err)
+		assert.Equal(t, 0, count)
+	})
+	t.Run("Error - wrong bucket", func(t *testing.T) {
+		svc := NewService(repos, tmpfile.Name(), successHTTPclient, "/base/url", 1, "/endpoint", 1)
+		time.Sleep(RepoSleepDuration)
+		count, err := svc.GetCount("fake")
+		assert.Error(t, err)
+		assert.Equal(t, 0, count)
+	})
+}
+
+func TestServiceImpl_GetAllConcepts_Success(t *testing.T) {
+	svc := createTestService(t, 200, nil)
+	time.Sleep(RepoSleepDuration)
+	pv, err := svc.GetAllConcepts("topics")
+	var wg sync.WaitGroup
+	var res string
+	wg.Add(1)
+	go func(reader io.Reader, w *sync.WaitGroup) {
+		var err error
+		blob, err := ioutil.ReadAll(reader)
+		assert.NoError(t, err)
+		log.Infof("Got bytes: %v", string(blob[:]))
+		res = string(blob[:])
+		wg.Done()
+	}(&pv, &wg)
+	wg.Wait()
+	assert.NoError(t, err)
+	assert.Equal(t, GetAllConceptsResult, res)
+}
+
+func TestServiceImpl_GetAllConcepts_Error(t *testing.T) {
+	svc := createTestService(t, 200, nil)
+	time.Sleep(RepoSleepDuration)
+	pv, err := svc.GetAllConcepts("fake")
+	var wg sync.WaitGroup
+	var res string
+	wg.Add(1)
+	go func(reader io.Reader, w *sync.WaitGroup) {
+		var err error
+		blob, err := ioutil.ReadAll(reader)
+		assert.NoError(t, err)
+		log.Infof("Got bytes: %v", string(blob[:]))
+		res = string(blob[:])
+		wg.Done()
+	}(&pv, &wg)
+	wg.Wait()
+
+	assert.NoError(t, err)
+	assert.Empty(t, res)
+}
+
+func TestServiceImpl_GetConceptUUIDs_Success(t *testing.T) {
+	svc := createTestService(t, 200, nil)
+	time.Sleep(RepoSleepDuration)
+	pv, err := svc.GetConceptUUIDs("topics")
+	var wg sync.WaitGroup
+	var res string
+	wg.Add(1)
+	go func(reader io.Reader, w *sync.WaitGroup) {
+		var err error
+		blob, err := ioutil.ReadAll(reader)
+		assert.NoError(t, err)
+		log.Infof("Got bytes: %v", string(blob[:]))
+		res = string(blob[:])
+		wg.Done()
+	}(&pv, &wg)
+	wg.Wait()
+	assert.NoError(t, err)
+	assert.Equal(t, GetConceptUUIDsResult, res)
+}
+
+func TestServiceImpl_GetConceptByUUID(t *testing.T) {
+	svc := createTestService(t, 200, nil)
+	time.Sleep(RepoSleepDuration)
+
+	t.Run("Success", func(t *testing.T) {
+		bc, found, err := svc.GetConceptByUUID("topics", "14fa0405-c625-3061-a1a0-a00643fc073f")
+
+		assert.NoError(t, err)
+		assert.True(t, found)
+		assert.Equal(t, "Fred", bc.PrefLabel)
+	})
+
+	t.Run("Error - bad type", func(t *testing.T) {
+		bc, found, err := svc.GetConceptByUUID("fake", "14fa0405-c625-3061-a1a0-a00643fc073f")
+
+		assert.Error(t, err)
+		assert.False(t, found)
+		assert.Empty(t, bc.PrefLabel)
+	})
+
+	t.Run("Error - wrong uuid", func(t *testing.T) {
+		bc, found, err := svc.GetConceptByUUID("topics", "12345678-c625-3061-a1a0-a00643fc073f")
+
+		assert.NoError(t, err)
+		assert.False(t, found)
+		assert.Empty(t, bc.PrefLabel)
+	})
+}
+
+func TestServiceImpl_SendConcepts(t *testing.T) {
```

Comment:
Is there any way of testing the failure to send concepts to S3, i.e. when response code is not 200? The SendConcepts method is very complicated and I am not entirely sure how it would behave on error conditions.